src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -42,7 +42,7 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
             DecorateOperationGroup();
         }
 
-        public IEnumerable<Resource> ArmResource => EnsureArmResource().Values;
+        public Dictionary<OperationGroup, Resource> ArmResource => EnsureArmResource();
```

Comment:
I prefer if we keep this IEnumerable and simply expose the operationGroup on the ArmResource object as an internal getter.  This keeps the context in place and allows all the writers to access both the object as well as its underlying operation group

Reply:
Done

Reply:
As [current V3 branch](https://github.com/Azure/autorest.csharp/blob/f727cd0d071cbb501d176d3407185ccfaee98ff6/src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs#L26) use OperarionGroup as key, which apporach should we take now?